### DISCUSSION TITLE
Cross Reference to table is not working. Getting warning unable to resolved.

### DISCUSSION THREAD

**Discussion by Hsotuhsa-S** [quarto-dev/quarto-cli#6311] [Q&A] [answered]
### Description

format: 
  pdf:

This Does not works. Cross reference is listed in 'insert Cross Reference' window.
But when render, warning message occurred "WARNING: Unable to resolve crossref @tbl-missing".
Render PDF display "See ?@tbl-missing" instead of reference to Table.

What am I missing here? 
Quarto version is 1.3.433.

See @tbl-missing

```{r missingValue}

#| label: tbl-missing
#| tbl-cap: Missing Values

# Count total missing values in each variable of data frame.
missing_values = cleanThyroidData_1 |>
                 summarise(across(everything(), ~ sum(is.na(.)))) |>
                 t()
# Print 
kable(missing_values, booktabs = T, align='c',
    caption = "Missing Values")|>
    kable_styling(latex_options = c("striped", "HOLD_position"))

```

Render output Warning 
"WARNING: Unable to resolve crossref @tbl-missing
running xelatex - 1
  This is XeTeX, Version 3.141592653-2.6-0.999995 (TeX Live 2023) (preloaded format=xelatex)
   restricted \write18 enabled.
  entering extended mode"

**Comment by mcanouil** [comment]
Could you share a small self-contained "working" (reproducible) example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Comment by Hsotuhsa-S** [comment]
Here is the output in pdf for Figure reference:
See Plot?@fig-plot

Here is the output in pdf for Table reference:
See?@tbl-iris

Here is example code.

````qmd
---
title: "Quarto Document"
format: pdf
---

See Plot @fig-plot

```{r}

#| label: fig-plot
#| fig-cap: Iris Plot

plot(iris$Sepal.Length)
```

See @tbl-iris

```{r tableRef}

#| label: tbl-iris
#| tbl-cap: Table Iris

iris = head(iris)

# Print 
knitr::kable(iris, booktabs = T, align='c',
    caption = "Table Iris")|>
    kable_styling(latex_options = c("striped", "HOLD_position"))

```

**Reply by mcanouil** [reply]
This is because your table code cell contains two labels, one inline (_i.e._, `tableRef`) and one as yaml (_i.e._, `tbl-iris`).
Also YAML option should be straight below the opening code cell, no empty lines.
Finally, don't use both `tbl-cap: Table Iris` and `caption = "Table Iris"`.

**Comment by Hsotuhsa-S** [comment]
What do you mean by "This is because your table code cell contains two labels, one inline (i.e., tableRef) and one as yaml (i.e., tbl-iris)." 
tableRef is chunk label. It is not a table label.

 "Finally, don't use both tbl-cap: Table Iris and caption = "Table Iris"." Are they are related?
Could you please elaborate these point? These information is not found in current reference document.

**Comment by Hsotuhsa-S** [comment]
1. Table is working when without caption. Following Is the test. 
But With caption getting rendering error.  Confusing.

See Table @tbl-Test

```{r table}
#| label: tbl-Test
#| tbl-cap: Table Iris

iris = head(iris)

# Print 
knitr::kable(iris, booktabs = T, align='c')|>
    kable_styling(latex_options = c("striped", "HOLD_position"))

```

2. Following is not working with 'caption'. Here is rending error "
compilation failed- error
Extra }, or forgotten \endgroup.
\@endfloatbox ...pagefalse \outer@nobreak \egroup 
                                                  \color@endbox 
l.275 \end{table}
see Test.log for more information."

See Table @tbl-Test

```{r table}
#| label: tbl-Test
#| tbl-cap: Table Iris

iris = head(iris)

# Print 
knitr::kable(iris, booktabs = T, align='c', caption = "irisHead")|>
    kable_styling(latex_options = c("striped", "HOLD_position"))

```
--
3. Cross reference Fig is same results as original post. 
   output is "See?@fig-Test"
   Rendering warning occurred. 
"WARNING: Unable to resolve crossref @fig-Test
running xelatex - 1
  This is XeTeX, Version 3.141592653-2.6-0.999995 (TeX Live 2023) (preloaded format=xelatex)
   restricted \write18 enabled.
  entering extended mode"


See @fig-Test

```{r plot}
#| lable: fig-Test
#| fig-cap: Plot Iris

plot(iris$Sepal.Length)

```

**Reply by mcanouil** [reply, marked as answer]
Everything works just fine if you follow the documentation about figure/table cross-reference and don't mixed inline/multiline/yaml style syntax for code cell options.

See <https://quarto.org/docs/authoring/cross-references.html> and <https://yihui.org/en/2022/01/knitr-news/>.

**Reply by Hsotuhsa-S** [reply]
Hey Thanks. Not sure what you mean with mutiline and YAML syntax in quadro ? could you please give example

**Reply by mcanouil** [reply]
Look at the links I shared.

**Reply by mcanouil** [reply]
@renatovargas  Sorry, but what's the point of your message on a one year old post? It's not helpful or constructive. Please refrain of posting such content which goes against the Code of Conduct.

The links explained everything way better than what I could have done and was marked as the answer by the user.

**Comment by bjornasm** [comment]
Edit: Not sure if this really was the source and solution of my problem.

~~I came to this issue due to having my own problems with referencing images from code blocks. Just for the help of others that might have the same issue as me (which was tangent to this one), remember to add your~~ `label` and `fig-cap`  ~~at the top of the code block.~~ 

~~This caused issues to me:~~
```python
#| eval: true
#| label: fig-myfig
#| fig-cap: This is my figure

plt.plot([1, 2, 3, 4])
plt.ylabel('some numbers')
plt.show()
```

But this worked:
```python
#| label: fig-myfig
#| fig-cap: This is my figure
#| eval: true

plt.plot([1, 2, 3, 4])
plt.ylabel('some numbers')
plt.show()
```

**Reply by cderv** [reply]
This seems odd to me that options order matters. Can you share a full reproducible example with me of what did not work ? Thanks !

**Reply by bjornasm** [reply]
Weird, I can't reproduce it right now. I'll try to invoke the same problem as when I encountered it - it may be something else that didn't worked as it should, the figure was displaying and the reference did not work. Maybe it was something with the evaluation.. I'll look into it and edit my question with a ~~strikethrough~~ until I find out.
